src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/innerassignment/InputInnerAssignmentSwitchAndSwitchExpression.java

```diff
@@ -0,0 +1,85 @@
+/*
+InnerAssignment
+
+
+*/
+package com.puppycrawl.tools.checkstyle.checks.coding.innerassignment;
+
+//non-compiled with javac: Compilable with Java14
+public class InputInnerAssignmentSwitchAndSwitchExpression {
+
+    public void test1(int mode) {
+        int x = 0;
+        switch(mode) {
+            case 2 -> {
+                x = 2;
+            }
+            case 1 -> x = 1;
+        }
+    }
+
+    public void test2(int mode) {
+        int x = 0;
+        switch(mode) {
+            case 2, 4, 6 -> {
+                x = 2;
```

Comment:
Create another test method like:
````
   public void test2(int mode) {
        int x = 0;
        switch(mode) {
            case 2, 4, 6 -> {
                x = y = 2;
````

I would expect a violation here.

Reply:
added in the same method